patches/tModLoader/Terraria/ModLoader/UI/Interface.cs

```diff
@@ -153,12 +155,33 @@ internal static class Interface
 						});
 					Main.SaveSettings();
 				}
-				else if (!ModLoader.DetectedModChangesForInfoMessage) { // Keep this at the end of the if/else chain since it doesn't necessarily change Main.menuMode
+				else if (!ModLoader.DownloadedDependenciesOnStartup) { // Keep this at the end of the if/else chain since it doesn't necessarily change Main.menuMode
+					// Find dependencies that need to be downloaded.
+					var deps = ModOrganizer.DownloadWorkshopDependencies().ToList();
+					bool showModChanges = !ModLoader.DetectedModChangesForInfoMessage;
+					bool promptDepDownloads = deps.Count != 0;
+					bool skip = !showModChanges && !promptDepDownloads;
+
+					ModLoader.DownloadedDependenciesOnStartup = true;
 					ModLoader.DetectedModChangesForInfoMessage = true;
 
-					string info = ModOrganizer.DetectModChangesForInfoMessage();
-					if (info != null)
-						infoMessage.Show(Language.GetTextValue("tModLoader.ShowNewUpdatedModsInfoMessage") + info, Main.menuMode);
+					if (!skip) {
+						string newDownloads = showModChanges ? ModOrganizer.DetectModChangesForInfoMessage() : null;
+						string dependencies = promptDepDownloads ? ModOrganizer.ListDependenciesToDownload(deps) : null;
+
+						if (promptDepDownloads)
+							ModLoader.DependenciesToDownload = deps;
```

Comment:
I'm not finding any reason why this is stored on ModLoader, nor needs to be. 
Remove?

Reply:
Was worried about local variable capturing, but it should probably be fine in hindsight.